runtime/src/deterministic.rs

```diff
@@ -747,15 +757,38 @@ impl crate::Spawner for Context {
 
         // Set up the task
         let executor = self.executor.clone();
+
         move |f: F| {
-            let (f, handle) = Handle::init_future(f, gauge, false);
+            // Give spawned task its own empty children list
+            let (f, handle) =
+                Handle::init_future(f, gauge, false, Arc::new(Mutex::new(Vec::new())));
 
             // Spawn the task
             Tasks::register_work(&executor.tasks, label, Box::pin(f));
             handle
         }
     }
 
+    fn spawn_child<F, Fut, T>(self, f: F) -> Handle<T>
+    where
+        F: FnOnce(Self) -> Fut + Send + 'static,
+        Fut: Future<Output = T> + Send + 'static,
+        T: Send + 'static,
+    {
+        // Store parent's children list
+        let parent_children = self.children.clone();
```

Comment:
While this may look like a lot of cloning, this is just behind an `Arc` (and its more efficient than some alternatives that avoid this call).